anylabeling/services/auto_labeling/sam_onnx.py

```diff
@@ -175,11 +178,21 @@ def encode(self, cv_image):
             (self.input_size[1], self.input_size[0]),
             flags=cv2.INTER_LINEAR,
         )
-
-        encoder_inputs = {
-            self.encoder_input_name: cv_image.astype(np.float32),
-        }
-        image_embedding = self.run_encoder(encoder_inputs)
+        
+        filename = filename.split(".")[0]
+        embedding_path = f"{filename}.npy"
+        
+        if os.path.exists(embedding_path):
+            image_embedding = np.load(embedding_path)
+        else:
+            self.encoder_session = onnxruntime.InferenceSession(
```

Comment:
If you initialize the model like that, the model will be loaded again and again each time the user calls `encode()`.